src/lib/datatypes/sol-buffer.c

```diff
@@ -868,3 +868,19 @@ sol_buffer_fini(struct sol_buffer *buf)
     buf->used = 0;
     buf->capacity = 0;
 }
+
+SOL_API struct sol_blob *
+sol_buffer_to_blob(struct sol_buffer *buf)
+{
+    struct sol_blob *blob;
+
+    SOL_NULL_CHECK(buf, NULL);
+
+    blob = sol_blob_new(SOL_BLOB_TYPE_DEFAULT, NULL, buf->data, buf->used);
```

Comment:
problem if you take the whole buffer into account for blob, this won't be useful for buffers allocated on the stack, that are pretty common. Taking only the buffer memory is more likely to be useful.

However I'd document this, that once you do the conversion, only the data is kept alive (and duplicated if needed), the buffer is empty (zeroed, data, used and capacity if not "memory_owned") and may be freed afterwards